ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -642,6 +649,45 @@ module.exports = function CesiumMap(
 
       return billboardRef
     },
+    /*
+     * Draws a label containing the text in the given options.
+     */
+    addLabel(point, options) {
+      const pointObject = convertPointCoordinate(point)
+      const cartographicPosition = Cesium.Cartographic.fromDegrees(
+        pointObject.longitude,
+        pointObject.latitude,
+        pointObject.altitude
+      )
+      const cartesianPosition = map.scene.globe.ellipsoid.cartographicToCartesian(
+        cartographicPosition
+      )
+      // X, Y offset values for the label
+      const offset = new Cesium.Cartesian2(20, -15)
+      // Cesium measuremenmt for determining how to render the size of the label based on zoom
+      const scaleZoom = new Cesium.NearFarScalar(1.5e4, 1.0, 8.0e6, 0.0)
+      // Cesium measurement for determining how to render the translucency of the label based on zoom
+      const translucencyZoom = new Cesium.NearFarScalar(1.5e6, 1.0, 8.0e6, 0.0)
+
+      const labelRef = labelCollection.add({
+        text: options.text,
+        position: cartesianPosition,
+        id: options.id,
+        pixelOffset: offset,
+        scale: 0.5,
+        scaleByDistance: scaleZoom,
+        translucencyByDistance: translucencyZoom,
+        fillColor: Cesium.Color.BLACK,
+        outlineColor: Cesium.Color.WHITE,
+        outlineWidth: 2,
+        style: Cesium.LabelStyle.FILL_AND_OUTLINE,
+        showBackground: true,
+        backgroundColor: Cesium.Color.WHITE,
+      })
+      mapModel.addLabel(labelRef)
```

Comment:
:question: Why are we calling `mapModel.addLabel` in cesium, but not in openlayers?

Reply:
OpenLayers actually has an option to turn on label decluttering, but Cesium doesn't. So I created that method to assist with searching for other labels with the same position in the Cesium implementation.